test/0.8.9/withdrawal-queue.test.js

```diff
@@ -1,30 +1,30 @@
-const hre = require('hardhat')
-const { artifacts, contract, ethers } = require('hardhat')
-const { bn, getEventArgument, ZERO_ADDRESS } = require('@aragon/contract-helpers-test')
+const { artifacts, contract, ethers, web3 } = require('hardhat')
+const { bn, getEventArgument, ZERO_ADDRESS, ZERO_BYTES32 } = require('@aragon/contract-helpers-test')
 
-const { ETH, StETH, shareRate, shares, setBalance } = require('../helpers/utils')
+const { ETH, StETH, shareRate, shares } = require('../helpers/utils')
 const { assert } = require('../helpers/assert')
-const withdrawals = require('../helpers/withdrawals')
 const { signPermit, makeDomainSeparator } = require('../0.6.12/helpers/permit_helpers')
 const { MAX_UINT256, ACCOUNTS_AND_KEYS } = require('../0.6.12/helpers/constants')
-const { impersonate, EvmSnapshot } = require('../helpers/blockchain')
+const { impersonate, EvmSnapshot, getCurrentBlockTimestamp, setBalance } = require('../helpers/blockchain')
 
-const StETHMock = artifacts.require('StETHMock.sol')
-const WstETH = artifacts.require('WstETHMock.sol')
+const { deployWithdrawalQueue } = require('./withdrawal-queue-deploy.test')
 
-contract('WithdrawalQueue', ([owner, stranger, daoAgent, user]) => {
+contract('WithdrawalQueue', ([owner, stranger, daoAgent, user, pauser, resumer, bunkerReporter]) => {
   let withdrawalQueue, steth, wsteth
 
   const snapshot = new EvmSnapshot(ethers.provider)
 
   before('Deploy', async () => {
-    steth = await StETHMock.new({ value: ETH(1) })
-    wsteth = await WstETH.new(steth.address)
-
-    withdrawalQueue = (await withdrawals.deploy(daoAgent, wsteth.address)).queue
+    const deployed = await deployWithdrawalQueue({
+      stethOwner: owner,
+      queueAdmin: daoAgent,
+      queuePauser: daoAgent,
+      queueResumer: daoAgent,
+    })
 
-    await withdrawalQueue.initialize(daoAgent, daoAgent, daoAgent, steth.address, steth.address)
-    await withdrawalQueue.resume({ from: daoAgent })
+    steth = deployed.steth
+    wsteth = deployed.wsteth
+    withdrawalQueue = deployed.withdrawalQueue
 
     await steth.setTotalPooledEther(ETH(600))
     // we need 1 ETH additionally to pay gas on finalization because coverage ingnores gasPrice=0
```

Comment:
if using a separate finalizer, this extra ETH on the stETH address won't be needed

Reply:
Yeah, that makes sense. We have talked about this but decided to keep things as is while we are working on additional tests so not to break existing ones.  
When finalizing this PR, I will see if we can sort it out.

Reply:
Yeah, good idea, basically.